store/etcdv3/etcdv3.go

```diff
@@ -0,0 +1,502 @@
+package etcdv3
+
+import (
+	"context"
+	"crypto/tls"
+	"strings"
+	"sync"
+	"time"
+
+	etcd "github.com/coreos/etcd/clientv3"
+	"github.com/coreos/etcd/clientv3/concurrency"
+	"github.com/docker/libkv"
+	"github.com/docker/libkv/store"
+)
+
+const (
+	defaultLockTTL     = 20 * time.Second
+	etcdDefaultTimeout = 5 * time.Second
+)
+
+// EtcdV3 is the receiver type for the
+// Store interface
+type EtcdV3 struct {
+	client *etcd.Client
+}
+
+type etcdLock struct {
+	lock  sync.Mutex
+	store *EtcdV3
+
+	mutex   *concurrency.Mutex
+	session *concurrency.Session
+
+	mutexKey string // mutexKey is the key to write appended with a "_lock" suffix
+	writeKey string // writeKey is the actual key to update protected by the mutexKey
+	value    string
+	ttl      time.Duration
+}
+
+// Register registers etcd to libkv
+func Register() {
+	libkv.AddStore(store.ETCDV3, New)
+}
+
+// New creates a new Etcd client given a list
+// of endpoints and an optional tls config
+func New(addrs []string, options *store.Config) (store.Store, error) {
+	s := &EtcdV3{}
+
+	var (
+		entries []string
+		err     error
+	)
+
+	entries = store.CreateEndpoints(addrs, "http")
+	cfg := &etcd.Config{
+		Endpoints: entries,
+	}
+
+	// Set options
+	if options != nil {
+		if options.TLS != nil {
+			setTLS(cfg, options.TLS, addrs)
+		}
+		if options.ConnectionTimeout != 0 {
+			setTimeout(cfg, options.ConnectionTimeout)
+		}
+		if options.Username != "" {
+			setCredentials(cfg, options.Username, options.Password)
+		}
+		if options.SyncPeriod != 0 {
```

Comment:
`SyncPeriod` is not in `docker/libkv/store.Config`. It's only in your fork. The import statement should be changed IMHO.

Reply:
Thanks, you're right. I'm still working on the main repository, pushing to my remote. I don't like renaming import paths (kinda dirty), but I don't think these features will end up being included upstream anyways. So I'll have to do it eventually 😞

(I also thought about creating a new repository with a new name to clearly mark a separation from the upstream libkv, I'm not sure yet)

Reply:
Oh and another solution: people could still reference the fork with their dependency manager (glide, etc.).

It's still one more step to contribute though: you have to add my remote and rebase on top of my master branch before submitting a PR with a feature/patch to the fork. I can write some docs for this though.

Reply:
> I also thought about creating a new repository with a new name to clearly mark a separation from the upstream libkv, I'm not sure yet

From my POV that would be the best solution. I would be interested in contributing. Right now it's not clear if it's temporary fork, if the changes will be merged to upstream repo. Based on the number of issues in upstream repo looks like there is high demand for that lib and for improving it.

> Oh and another solution: people could still reference the fork with their dependency manager (glide, etc.).

That's true but only if you use package managers that support that. I've recently migrated to [dep](https://github.com/golang/dep) because of significant speed improvements and dep doesn't have that feature.

Reply:
Thank you for the feedback. I'll certainly take that into account. I will keep on working on it and prepare for a rename if needs be. I have no intention to contribute to the upstream repo because it does not seem active anymore, so this fork is actually where the work will be happening. I guess a rename makes sense in this case.

Reply:
One question. Why have you disabled Issues in your fork?

Reply:
I haven't disabled them, I forgot to turn them on 🤣 I thought they were enabled by default on forks somehow, seems not. Thanks for pointing that out.